src/rust/lib/system/web/src/lib.rs

```diff
@@ -573,6 +574,28 @@ pub fn forward_panic_hook_to_console() {
     console_error_panic_hook::set_once();
 }
 
+
+/// Enables throwing a descriptive JavaScript error on panics.
+pub fn forward_panic_hook_to_error() {
+    panic::set_hook(Box::new(error_throwing_panic_hook));
+}
+
+#[wasm_bindgen(module = "/js/rust_panic.js")]
+extern "C" {
+    fn new_panic_error(message:String) -> JsValue;
+}
```

Comment:
I'd use here inline JS instead. This would be easier for the reader to track whats going on, especially as the JS file is so small. But Im also completel ok with how it looks now, so its up to your preference :)

Reply:
I do prefer it in a separate file because it gives you IDE support. Inline, I don't even get syntax highlighting.